### ISSUE TITLE
Periodic asyncio.CancelledError from httpx

### ISSUE THREAD

**Issue by TheTechromancer** [blacklanternsecurity/bbot#632]
This bug is causing modules to crash and the overall scan to hang.

```
Traceback (most recent call last):
  File "/root/bbot/bbot/modules/base.py", line 353, in _worker
    await self.handle_event(event)
  File "/root/bbot/bbot/modules/subdomain_hijack.py", line 44, in handle_event
    hijackable, reason = await self.check_subdomain(event)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/bbot/bbot/modules/subdomain_hijack.py", line 90, in check_subdomain
    match, reason = await self._verify_fingerprint(f, url, headers=headers)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/bbot/bbot/modules/subdomain_hijack.py", line 102, in _verify_fingerprint
    r = await self.helpers.request(*args, **kwargs)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/bbot/bbot/core/helpers/web.py", line 116, in request
    response = await client.request(*args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/.cache/pypoetry/virtualenvs/bbot-aKeTQkGo-py3.11/lib/python3.11/site-packages/httpx/_client.py", line 1530, in request
    return await self.send(request, auth=auth, follow_redirects=follow_redirects)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/.cache/pypoetry/virtualenvs/bbot-aKeTQkGo-py3.11/lib/python3.11/site-packages/httpx/_client.py", line 1617, in send
    response = await self._send_handling_auth(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/.cache/pypoetry/virtualenvs/bbot-aKeTQkGo-py3.11/lib/python3.11/site-packages/httpx/_client.py", line 1645, in _send_handling_auth
    response = await self._send_handling_redirects(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/.cache/pypoetry/virtualenvs/bbot-aKeTQkGo-py3.11/lib/python3.11/site-packages/httpx/_client.py", line 1682, in _send_handling_redirects
    response = await self._send_single_request(request)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/.cache/pypoetry/virtualenvs/bbot-aKeTQkGo-py3.11/lib/python3.11/site-packages/httpx/_client.py", line 1719, in _send_single_request
    response = await transport.handle_async_request(request)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/.cache/pypoetry/virtualenvs/bbot-aKeTQkGo-py3.11/lib/python3.11/site-packages/httpx/_transports/default.py", line 353, in handle_async_request
    resp = await self._pool.handle_async_request(req)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/.cache/pypoetry/virtualenvs/bbot-aKeTQkGo-py3.11/lib/python3.11/site-packages/httpcore/_async/connection_pool.py", line 262, in handle_async_request
    raise exc
  File "/root/.cache/pypoetry/virtualenvs/bbot-aKeTQkGo-py3.11/lib/python3.11/site-packages/httpcore/_async/connection_pool.py", line 245, in handle_async_request
    response = await connection.handle_async_request(request)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/.cache/pypoetry/virtualenvs/bbot-aKeTQkGo-py3.11/lib/python3.11/site-packages/httpcore/_async/connection.py", line 69, in handle_async_request
    stream = await self._connect(request)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/.cache/pypoetry/virtualenvs/bbot-aKeTQkGo-py3.11/lib/python3.11/site-packages/httpcore/_async/connection.py", line 117, in _connect
    stream = await self._network_backend.connect_tcp(**kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/.cache/pypoetry/virtualenvs/bbot-aKeTQkGo-py3.11/lib/python3.11/site-packages/httpcore/_backends/auto.py", line 31, in connect_tcp
    return await self._backend.connect_tcp(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/.cache/pypoetry/virtualenvs/bbot-aKeTQkGo-py3.11/lib/python3.11/site-packages/httpcore/_backends/anyio.py", line 114, in connect_tcp
    stream: anyio.abc.ByteStream = await anyio.connect_tcp(
                                   ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/.cache/pypoetry/virtualenvs/bbot-aKeTQkGo-py3.11/lib/python3.11/site-packages/anyio/_core/_sockets.py", line 221, in connect_tcp
    await event.wait()
  File "/root/.cache/pypoetry/virtualenvs/bbot-aKeTQkGo-py3.11/lib/python3.11/site-packages/anyio/_backends/_asyncio.py", line 1778, in wait
    if await self._event.wait():
       ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/asyncio/locks.py", line 213, in wait
    await fut
asyncio.exceptions.CancelledError
```

**Comment by TheTechromancer** [comment]
This appears to be a bug in httpx/httpcore. Opened a discussion here:

https://github.com/encode/httpx/discussions/2792

**Comment by TheTechromancer** [comment]
Might also be a bug in AnyIO. From the traceback it is impossible to tell. Heaven save us all.

**Comment by TheTechromancer** [comment]
Discussion on AnyIO:

https://github.com/agronholm/anyio/issues/534

From the creator of AnyIO:

> Concurrency in programming is hard. We're trying to make it less painful (structured concurrency helps a lot), but we still have ways to go.

> As far as your traceback goes, it's hard to say where this cancellation comes from, but it doesn't seem to come from AnyIO, as the exception would not fall out of a task. If you want to be sure, try with AnyIO 4.0 (you should see a message associated with the exception if, by some chance, an AnyIO exception did in fact fall out of a task). It's a good idea to switch anyway given the myriad cancellation related fixes that went into the release. Best of luck!

**Comment by TheTechromancer** [comment]
Finally was able to reproduce this with the following code:

```python
import httpx
import random
import asyncio

## STEP 1:
##   python -m http.server 5555

async def main():
    async def request():
        transport = httpx.AsyncHTTPTransport(retries=5) # <-----------------------
        async with httpx.AsyncClient(transport=transport, timeout=random.random() / 10) as client:
            try:
                return await client.get("http://127.0.0.1:5555")
            except httpx.TimeoutException:
                print("_", end="", flush=True)

    while 1:
        tasks = [asyncio.create_task(request()) for _ in range(100)]
        await asyncio.gather(*tasks)

asyncio.run(main())
```

Confirmed this was an AnyIO bug, and that's been fixed in 4.0.0rc1. Fix pushed to `misc_cleanup`:

https://github.com/blacklanternsecurity/bbot/pull/607/commits/5112bebd803866889931c4592079811a02bbc27f
